addons/mail/static/src/models/thread/thread.js

```diff
@@ -328,12 +328,17 @@ function factory(dependencies) {
                 method: 'channel_join_and_get_info',
                 args: [[channelId]]
             });
-            const thread = this.insert(Object.assign(
-                { isPinned: true },
-                this.convertData(data)
-            ));
+            const convertedData = Object.assign(
+                { pendingPinState: undefined },
```

Comment:
This feels not enough, `pendingPinState` should be cleaned only if new value is actually equal to expected value (or even better, only if the notif is a direct result of the rpc that requested it), and it should ideally be for any update and not be flux specific.

I already made a similar comment in earlier days with this conceptual issue.